File: 2565. Subsequence With the Minimum Score/Solution.py
class Solution:
    def minimumScore(self, s: str, t: str) -> int:
        m, n = len(s), len(t)
        left, right = 0, n - 1
        dp = [0 for i in range(m)]
        for i in range(m):
            if left < n and s[i] == t[left]: left += 1
            dp[i] = left
        res = n - left
        for i in range(m - 1, -1, -1):
            res = min(res, max(0, right - dp[i] + 1))
            if right > 0 and s[i] == t[right]: right -= 1
        return min(res, right + 1)

File: 2565. Subsequence With the Minimum Score/Solution.java
class Solution {
    public int minimumScore(String s, String t) {
        int m = s.length(), n = t.length(), left = 0, right = n - 1;
        int[] dp = new int[m];
        for (int i = 0; i < m; i++) {
            if (left < n && s.charAt(i) == t.charAt(left)) left++;
            dp[i] = left;
        }
        int res = n - left;
        for (int i = m - 1; i >= 0; i--) {
            res = Math.min(res, Math.max(0, right - dp[i] + 1));
            if (right > 0 && s.charAt(i) == t.charAt(right)) right--;
        } 
        return Math.min(res, right + 1);
    }
}